torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3519,7 +3627,7 @@ def _get_default_cuda_device(module: nn.Module) -> torch.device:
     except StopIteration:
         pass
     # Fall back to current CUDA device
-    return torch.device("cuda")
+    return torch.device("cuda", torch.cuda.current_device())
```

Comment:
@zhaojuanmao I think this change should be okay,  this makes it truly the current cuda device, instead of the device just being "cuda", it will be like "cuda:0" or "cuda:1", but can you confirm this is ok?

Reply:
sounds good, thanks!